app/src/main/java/org/schabi/newpipe/fragments/detail/VideoDetailFragment.java

```diff
@@ -779,6 +811,14 @@ public void onLoadingFailed(final String imageUri, final View view,
         }
     }
 
+    private void showHideKodiButton() {
+        // show kodi button if it supports the current service and it is enabled in settings
+        final boolean showKodiButton = KoreUtil.isServiceSupportedByKore(serviceId)
+                && PreferenceManager.getDefaultSharedPreferences(requireContext())
+                .getBoolean(getString(R.string.show_play_with_kodi_key), false);
```

Comment:
You can use `shouldShowPlayWithKodi()` now:
https://github.com/TeamNewPipe/NewPipe/blob/8fd48a88be33dc09a7215ef31044922ea7c7e376/app/src/main/java/org/schabi/newpipe/util/KoreUtil.java#L19